scripts/build/generator/generate-2023-auto-theme.mjs

```diff
@@ -0,0 +1,166 @@
+import * as fs from "node:fs";
+import * as path from 'node:path';
+
+import * as svg from "svgson";
+
+import { __dirname } from "../constants.mjs";
+
+function removeExtension(fileName) {
+    return path.basename(fileName, path.extname(fileName))
+}
+
+function removeSuffix(fileName) {
+    return removeExtension(fileName).split('_')[0]
+}
+
+function getIconName(fileName) {
+    return removeSuffix(removeExtension(fileName))
+}
+
+/**
+ * @param {import("../types/Theme").Theme} lightTheme The light theme.
+ * @param {import("../types/Theme").Theme} darkTheme The dark theme.
+ * @returns {Record<string, { light: svg.INode, dark: svg.INode?, iconName: string }>} The icons.
+*/
+function getIcons(lightTheme, darkTheme) {
+    const icons = {}
+
+    const iconDefinitions = [
+        ...Object.entries(lightTheme.iconDefinitions).map(([iconDefinitionKey, iconDefinition]) => /** @type {const} */ ([iconDefinitionKey, { ...iconDefinition, theme: 'light' }])),
+        ...Object.entries(darkTheme.iconDefinitions).map(([iconDefinitionKey, iconDefinition]) => /** @type {const} */ ([iconDefinitionKey, { ...iconDefinition, theme: 'dark' }])),
+    ]
+
+    for (const [iconDefinitionKey, iconDefinition] of iconDefinitions) {
+        if (!fs.existsSync(iconDefinition.iconPath)) {
+            continue
+        }
+
+        const iconName = getIconName(path.basename(iconDefinition.iconPath))
+
+        if (!icons[iconDefinitionKey]) {
+            icons[iconDefinitionKey] = {
+                light: null,
+                dark: null,
+                iconName,
+            }
+        }
+
+        const iconSource = fs.readFileSync(iconDefinition.iconPath, { encoding: 'utf8' })
+        const iconAst = svg.parseSync(iconSource)
+
+        icons[iconDefinitionKey][iconDefinition.theme] = iconAst
+    }
+
+    return /** @type {any} */ (icons)
```

Comment:
I would recommend using TypeScript instead of typing with JSDoc. JSDoc is better suited for when you're building a library.

Also this may be subjective but personally not a big fan of the JSDoc syntax. TypeScript seems much cleaner and easier to  read.

If you convert this to TypeScript we can use the [ts-node](https://www.npmjs.com/package/ts-node) package to run the typescript file on `prebuild`

Reply:
I used `/** @type {any} */` to prevent VSCode from marking an error and because I am sure that it matches the required return type. It can be safely removed.

I did not want to not add new dependencies to run Typescript (it is not really necessary), but I wanted some type safety because the functions are a bit complex.

I don't think this is a problem, but if you really think it is, we can also remove the JSDoc comments. If you want to use Typescript, I would recommend using [tsx](https://github.com/esbuild-kit/tsx) (see the [comparisons](https://github.com/privatenumber/ts-runtime-comparison) with ts-node) to run the build script.

Reply:
If you don't mind using TypeScript then let's use that. Feel free to use tsx though.